script.js

```diff
@@ -1,12 +1,207 @@
+
+//make card for each episode
 function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  let output = "";
+  episodeList.forEach((episode) => {
+    output += `
+      <div class="col">
+        <div class="card h-100 px-0 mx-1">
+        <img src="${episode.image.medium}" class="card-img-top mx-0" alt="...">
+        <div class="card-body mb-4">
+        <h5 >${episode.name} - S${episode.season
+      .toString()
+      .padStart(2, "0")}E${episode.season.toString().padStart(2, "0")}</h5>
+      <p>${
+        episode.summary.split(" ").length >= 25
+          ? episode.summary.split(" ").slice(0, 24).join(" ").concat("...")
+          : episode.summary.split(" ").slice(0, 24).join(" ")
+      }</p>
+        </div>
+        <a href="${
+          episode.url
+        }" class="btn btn-primary" style="position:absolute; bottom:10px; left:20px; right:20px;" target="_blank">See More</a>
+      </div>
+      </div>
+    `;
+    showEpisodseRow.innerHTML = output;
+  });
+
+  // Create option for each episodes in select after loading page
+  episodeList.forEach((episode) => {
+    const option = document.createElement("option");
+    option.value = `S${episode.season
+      .toString()
+      .padStart(2, "0")}E${episode.number.toString().padStart(2, "0")}`;
+    option.textContent = `S${episode.season
+      .toString()
+      .padStart(2, "0")}E${episode.number.toString().padStart(2, "0")} - ${
+      episode.name
+    }`;
+    selectEpisodes.appendChild(option);
+  });
 }
 
-window.onload = setup;
+// select a specific episode
+function selectEpisode(event) {
+  if (event.target.value == "allEpisodes") {
+    searchUser.value = "";
+    searchUser.disabled = false;
+    //Remove options from episodes's select
+    const options = selectEpisodes.getElementsByTagName("option");
+    for (let i = options.length - 1; i > 0; i--) {
+      selectEpisodes.removeChild(options[i]);
+    }
+    makePageForEpisodes(allEpisodes);
+  } else {
+    searchUser.value = "";
+    searchUser.disabled = true;
+    const season = event.target.value.slice(1, 3);
+    const number = event.target.value.slice(4, 6);
+    const selectedMovie = allEpisodes.filter(
+      (episode) => episode.season == season && episode.number == number
+    );
+    makePageForEpisodes(selectedMovie);
+  }
+}
+
+// Show episodes based on live search
+function searchMovie(e) {
+  const selectedMovie = [];
+  // Get the value of input
+  let serchContent = e.target.value;
+  if (serchContent == "") {
+    makePageForEpisodes(allEpisodes);
+    searchNumResult.textContent = "";
+  } else {
+    // Create case-insensitive RegExp
+    let searchContentInsensitive = new RegExp(serchContent, "i");
+    allEpisodes.forEach((episode) => {
+      if (
+        episode.name.match(searchContentInsensitive) !== null ||
+        episode.summary.slice(0, 25).match(searchContentInsensitive) !== null
```

Comment:
How did you decide that only the first 25 characters should be searched for?

Reply:
> This generally looks good! I noticed one bug when testing our your page - if I enter a search filter in the text box which _doesn't_ match any episodes (e.g. "asdfgh"), it shows _all_ of the episodes rather than none. Can you work out why?

I will correct it. Thanks for your tip

Reply:
That's actually a really good reason to only search the first 25 characters! Good job thinking about what would be clear vs confusing to users! (I could definitely argue this either way).

One question though - is there a way you could make it more clear from the structure of your code this relationship?

Imagine if you started showing the first 50 characters instead of 25 - you would want to search over those full 50 characters, right? But it would be easy to forget to update the search function.

How could you structure your code to make it more likely that both will be changed at the same time?

Reply:
I corrected it.